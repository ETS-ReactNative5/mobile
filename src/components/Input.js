import React from 'react'
import {
  TextInput,
  View
} from 'react-native'
import StyledText from './StyledText'
import EStyleSheet from 'react-native-extended-stylesheet'

const Input = (props) => {
  return (
    <View>
      { props.labelText !== '' ? <StyledText text={props.labelText} /> : null }
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          onChangeText={props.handleOnChangeText}
          secureTextEntry={props.passwordField}
        />
      </View>
    </View>
  )
}

const styles = EStyleSheet.create({
  input: {
    backgroundColor: 'transparent',
    height: 40,
    padding: 8,
  },
  inputContainer: {
    borderColor: 'grey',
    borderWidth: 1,
    marginBottom: 10
  },
});

Input.propTypes = {
  handleOnChangeText: React.PropTypes.func,
  labelText: React.PropTypes.string,
  passwordField: React.PropTypes.bool
}

Input.defaultProps = { labelText: '', passwordField: false };

export default Input
