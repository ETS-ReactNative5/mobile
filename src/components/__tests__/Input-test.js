import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Input from '../Input'


it('renders with no label', () => {
  const tree = renderer.create(
    <Input />
  ).toJSON();
  expect(tree).toMatchSnapshot()
})

it('renders a label correctly', () => {
  const tree = renderer.create(
    <Input labelText={'Ermahgerd Berks'}/>
  ).toJSON();
  expect(tree).toMatchSnapshot()
})

it('renders password fields (using secureTextEntry)', () => {
  const tree = renderer.create(
    <Input passwordField={true}/>
  ).toJSON();
  expect(tree).toMatchSnapshot()
})
