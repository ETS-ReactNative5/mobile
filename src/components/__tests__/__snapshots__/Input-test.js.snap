exports[`test renders a label correctly 1`] = `
<View
  style={undefined}>
  <View
    style={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          undefined,
        ]
      }>
      Ermahgerd Berks
    </Text>
  </View>
  <View
    style={undefined}>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="default"
      onChangeText={undefined}
      returnKeyType="done"
      secureTextEntry={false}
      style={undefined}
      underlineColorAndroid="white" />
  </View>
</View>
`;

exports[`test renders password fields (using secureTextEntry) 1`] = `
<View
  style={undefined}>
  <View
    style={undefined}>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="default"
      onChangeText={undefined}
      returnKeyType="done"
      secureTextEntry={true}
      style={undefined}
      underlineColorAndroid="white" />
  </View>
</View>
`;

exports[`test renders with no label 1`] = `
<View
  style={undefined}>
  <View
    style={undefined}>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="default"
      onChangeText={undefined}
      returnKeyType="done"
      secureTextEntry={false}
      style={undefined}
      underlineColorAndroid="white" />
  </View>
</View>
`;
